src/utils/contract.ts

```diff
@@ -119,4 +120,37 @@ export default class Contract extends BaseConnection {
     });
     return await response.json();
   }
+
+  private async fetchDataViaProvider(params) {
+    switch (this.providerType) {
+      case "ethers":
+        return await this.web3Provider!
```

Comment:
There's some inconsistent behaviour: if `providerType` equals `"ethers"` and `web3Provider` is absent, JS will throw an exception; if `providerType` equals `"web3"` and `web3Provider` is absent, it'll be no-op. Either scenario is bad... We should validate these parameters on `EthereumNamingService` constructor's level instead